powerdns/views.py

```diff
@@ -92,6 +102,94 @@ def get_queryset(self):
 
 
+
+class GUIRecordViewSet(OwnerViewSet):
+
+    queryset = Record.objects.all().select_related('owner', 'domain')
+    serializer_class = RecordSerializer
+    filter_fields = ('name', 'content', 'domain')
+    search_fields = filter_fields
+
+    def get_serializer_class(self):
+        if self.request.method in ['POST', 'PUT', 'PATCH']:
+            serializer_class = RecordRequestSerializer
+        else:
+            serializer_class = RecordSerializer
+        print(serializer_class)
+        return serializer_class
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        # TODO:: this code needs some love, any idea?
+        auto_accept = (
+            self.request.user.is_superuser or
+            serializer.instance.domain.unrestricted == True or
+            self.request.user == serializer.instance.domain.owner or
+            request.user.id in serializer.instance.domain.authorisations.values_list('authorised', flat=True) # noqa
+        )
+        if auto_accept:
+            record = serializer.instance.accept()
+            serializer.instance.record = record
+            serializer.instance.save()
+            serializer.data['record'] = record
+            code = status.HTTP_201_CREATED
+            headers = {'Location': serializer.data['record']}
+        else:
+            code = status.HTTP_202_ACCEPTED
+            headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=code, headers=headers)
+
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', False)
+        instance = get_object_or_404(
+            RecordRequest, record__pk=self.get_object().pk)
+        serializer = self.get_serializer(
+            instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        auto_accept = (
+            self.request.user.is_superuser or
+            # serializer.instance.domain.unrestricted == True or
+            # self.request.user == serializer.instance.domain.owner or
+            request.user.id == instance.record.owner.id or
+            request.user.id in instance.record.authorisations.values_list(
+                'authorised', flat=True)
+        )
+        if auto_accept:
+            record = serializer.instance.accept()
+            serializer.instance.record = record
+            serializer.instance.save()
+            serializer.data['record'] = record
+            code = status.HTTP_200_OK
+            headers = {'Location': serializer.data['record']}
+        else:
+            code = status.HTTP_202_ACCEPTED
+            headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=code, headers=headers)
+
+    def get_queryset(self):
```

Comment:
this is the same as in the `RecordViewSet`. Maybe we could move it somewhere to common place? Maybe some Record manager? Then calling `Record.objects.filter(ip='1.2.3.4')` will make filter like here. Other thing is, that this logic (filtering) should be moved to some filter backend (maybe sth like http://www.django-rest-framework.org/api-guide/filtering/#specifying-a-filterset ?)